tests/__init__.py

```diff
@@ -116,8 +119,24 @@ async def _create_db():
     return db
 
 
+class _ActiveDirectoryLDAPServer(LDAPServer):
+    """LDAPServer implementation with Active Directory
+    specific attribute rootDomainNamingContext"""
+    def getRootDSE(self, request, reply):
+        root = interfaces.IConnectedLDAPEntry(self.factory)
+        reply(pureldap.LDAPSearchResultEntry(
+            objectName='',
+            attributes=[('supportedLDAPVersion', ['3']),
+                        ('namingContexts', [root.dn.getText()]),
+                        ('supportedExtension', [
+                            pureldap.LDAPPasswordModifyRequest.oid, ]),
+                        ('rootDomainNamingContext', ['DC=example,DC=org']), ], ))
+        return pureldap.LDAPSearchResultDone(
+            resultCode=ldaperrors.Success.resultCode)
+
+
 class _LDAPServerFactory(ServerFactory):
-    protocol = LDAPServer
+    protocol = _ActiveDirectoryLDAPServer
```

Comment:
Perhaps we should only do this for the AD-specific tests? (and the _ActiveDirectoryLDAPServer defintion could be moved to `test_ad.py` ?)